src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -557,16 +576,21 @@ exprt get_numeric_value_from_character(
   constant_exprt A_char=from_integer('A', char_type);
   constant_exprt ten_int=from_integer(10, char_type);
 
-  binary_relation_exprt upper_case(chr, ID_ge, A_char);
+  // There are four cases: lower case, upper case, a digit or '+' and '-'
```

Comment:
There were already unit tests for that function. I've updated them for the new functionality of returning 0 for '+' and '-'.